app/main.py

```diff
@@ -1,6 +1,44 @@
+import os
+
+import requests
+
+URL = "https://api.weatherapi.com/v1/current.json"
+
+API_KEY = os.getenv("API_KEY")
+
+PARAMS = {
+    "key": API_KEY,
+    "q": "Paris",
+}
+
+
+def get_res() -> dict:
+    if not API_KEY:
+        raise ValueError("API key is not set!!!")
+    try:
+        print("Performing request to Weather API to city Paris...")
+        response = requests.get(URL, params=PARAMS)
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.HTTPError as err:
+        print(f"HTTP ERROR: {err}")
+    except requests.exceptions.RequestException as err:
+        print(f"REQUEST ERROR: {err}")
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    weather = get_res()
+    if not weather:
+        raise ValueError("Data not received!")
```

Comment:
The `get_weather` function does not handle cases where `get_res()` might return `None` or an unexpected structure. Adding checks for expected data will improve robustness.